src/library/export/engineprimeexportjob.cpp

```diff
@@ -110,6 +110,55 @@ std::optional<djinterop::track> getTrackByRelativePath(
     }
 }
 
+bool tryGetBeatgrid(BeatsPointer pBeats,
+        mixxx::audio::FramePos cuePlayPos,
+        int64_t frameCount,
+        std::vector<djinterop::beatgrid_marker>* pBeatgrid) {
```

Comment:
The code in this method is essentially unchanged from before, but has been lifted out of the main `exportMetadata()` function, and early-exit conditions added if any positions turn out to be invalid.